Gems/AtomTressFX/Code/Passes/HairPPLLRasterPass.cpp

```diff
@@ -0,0 +1,315 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+//#include <Atom/RHI/CommandList.h>
+#include <Atom/RHI/RHISystemInterface.h>
+#include <Atom/RHI/DrawPacketBuilder.h>
+#include <Atom/RHI/PipelineState.h>
+
+#include <Atom/RPI.Public/View.h>
+#include <Atom/RPI.Public/RPIUtils.h>
+#include <Atom/RPI.Public/RenderPipeline.h>
+#include <Atom/RPI.Public/RPISystemInterface.h>
+#include <Atom/RPI.Public/Pass/PassUtils.h>
+#include <Atom/RPI.Public/Scene.h>
+
+#include <Atom/RPI.Reflect/Asset/AssetUtils.h>
+#include <Atom/RPI.Reflect/Pass/RasterPassData.h>
+#include <Atom/RPI.Reflect/Pass/PassTemplate.h>
+#include <Atom/RPI.Reflect/Shader/ShaderAsset.h>
+
+#include <Passes/HairPPLLRasterPass.h>
+#include <Rendering/HairRenderObject.h>
+#include <Rendering/HairFeatureProcessor.h>
+
+namespace AZ
+{
+    namespace Render
+    {
+        namespace Hair
+        {
+
+            // --- Creation & Initialization ---
+            RPI::Ptr<HairPPLLRasterPass> HairPPLLRasterPass::Create(const RPI::PassDescriptor& descriptor)
+            {
+                RPI::Ptr<HairPPLLRasterPass> pass = aznew HairPPLLRasterPass(descriptor);
+                return pass;
+            }
+
+            HairPPLLRasterPass::HairPPLLRasterPass(const RPI::PassDescriptor& descriptor)
+                : RasterPass(descriptor),
+                m_passDescriptor(descriptor)
+            {
+            }
+
+            HairPPLLRasterPass::~HairPPLLRasterPass()
+            {
+            }
+
+            bool HairPPLLRasterPass::AcquireFeatureProcessor()
+            {
+                if (m_featureProcessor)
+                {
+                    return true;
+                }
+
+                RPI::Scene* scene = GetScene();
+                if (scene)
+                {
+                    m_featureProcessor = scene->GetFeatureProcessor<HairFeatureProcessor>();
+                }
+                else
+                {
+                    return false;
+                }
+
+                if (!m_featureProcessor)
+                {
+                    AZ_Warning("Hair Gem", false,
+                        "HairPPLLRasterPass [%s] - Failed to retrieve Hair feature processor from the scene",
+                        GetName().GetCStr());
+                    return false;
+                }
+                return true;
+            }
+
+            void HairPPLLRasterPass::InitializeInternal()
+            {
+                if (GetScene())
```

Comment:
It's a good point but it can be and was in the past.  The reason is that the called inherited method is creating and setting features that might not yet be ready and so this is postponed until the scene is set.